ingestion/src/metadata/profiler/processor/core.py

```diff
@@ -603,7 +603,9 @@ def get_profile(self) -> CreateTableProfileRequest:
 
             if self._system_results:
                 system_profile = [
-                    SystemProfile(**system_result)
+                    system_result
+                    if isinstance(system_result, SystemProfile)
+                    else SystemProfile(**system_result)
```

Comment:
This seems like something we should refactor to have a standard return type i.e. SystemProfile

Reply:
do u mean `metric_type != MetricTypes.System.value` ?

Reply:
With pydantic better to use:
```suggestion
                    system_result
                    if isinstance(system_result, SystemProfile)
                    else SystemProfile.model_validate(system_result)
```

Reply:
so if metric_type = MetricTypes.System.value and still profile=None or [] it will still not convert profile into {} on line `395`